core/query.go

```diff
@@ -16,16 +16,24 @@ import (
 	"github.com/moby/buildkit/util/leaseutil"
 	"github.com/opencontainers/go-digest"
 	"github.com/vektah/gqlparser/v2/ast"
+	"github.com/vito/progrock"
 )
 
 // Query forms the root of the DAG and houses all necessary state and
 // dependencies for evaluating queries.
 type Query struct {
+	QueryOpts
 	Buildkit *buildkit.Client
 
 	// The current pipeline.
+	// TODO:(XXX) should this be shared? was the previous behavior technically
```

Comment:
cc @vito it seems like previously `Pipeline` was shared across all clients attached to a given `DaggerServer`, was that intentional? It seems like that would result in them overriding each other unexpectedly (though I admittedly didn't really trace very deep through these codepaths, so low confidence). Either way, this change results in it being initialized per-client, let me know if you see any problems with that.

Reply:
Yep - whenever a pipeline was added the `Query` could be cloned and the appended path would be set on the clone, so this was more dependent on query chaining than per-client state. Any change here should be fine, we're not relying on this pipeline stuff for the new visualization stack anyway.